admin/services/marking.service.js

```diff
@@ -19,10 +20,12 @@ markingService.batchMark = async function (batchIds) {
   }
 
   const completedChecks = await completedCheckDataService.sqlFindByIds(batchIds)
-
+  const checkFormIds = completedChecks.map(c => c.checkForm_id)
+  const checkForms = await checkFormDataService.sqlFindByIds(checkFormIds)
   for (let cc of completedChecks) {
     try {
-      await this.mark(cc)
+      const checkForm = checkForms.find(cf => cc.checkForm_id === cf.id)
```

Comment:
Do we really want the overhead of a `find` here for every iteration of the loop?  It would be more efficient to use a hash map.

Reply:
Please move this lookup to use an object/properties approach.